### PR TITLE
Converts all attack animations to the new proper type

### PR DESCRIPTION
**Pull request by DominikPanic**
begone oldcode

### CODE REVIEW

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -476,18 +476,30 @@
...
-	if(direction & NORTH)
-		pixel_y_diff = 8
-	else if(direction & SOUTH)
-		pixel_y_diff = -8
-
-	if(direction & EAST)
-		pixel_x_diff = 8
-	else if(direction & WEST)
-		pixel_x_diff = -8
+	switch(direction)
```
Comment (spookydonut) [inline code/game/atoms_movable.dm]: this is worse code, change it back.

Comment (DominikPanic) [reply code/game/atoms_movable.dm]: Why is it worse?

Comment (spookydonut) [reply code/game/atoms_movable.dm]: because the old code treats them like bitflags instead of enums, and direction is bitflags not enums

Comment (Rohesie) [reply code/game/atoms_movable.dm]: This is actually better performant, and doing bitflag comparisons adds nothing. It's neither more clear nor more efficient.

**Pull request merged**